includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -367,23 +367,52 @@ public function get_product_data_for_subscription( WC_Subscription $subscription
 				continue;
 			}
 
-			try {
-				// TODO: Use the tax service get_tax_rates_for_item() once implemented.
-				$tax_rates = [];
-			} catch ( API_Exception $e ) {
-				return null;
-			}
-
 			$product_data[] = [
 				'price'     => $this->get_stripe_price_id( $product ),
 				'quantity'  => $item->get_quantity(),
-				'tax_rates' => $tax_rates,
+				'tax_rates' => $this->get_tax_rates_for_item( $item, $subscription ),
 			];
 		}
 
 		return $product_data;
 	}
 
+	/**
+	 * Prepare tax rates for a subscription item.
+	 *
+	 * @param WC_Order_Item   $item         Subscription order item.
+	 * @param WC_Subscription $subscription A Subscription to get tax rate information from.
+	 *
+	 * @return array
+	 */
+	public function get_tax_rates_for_item( WC_Order_Item $item, WC_Subscription $subscription ) {
+		$tax_rates = [];
+
+		if ( ! wc_tax_enabled() || ! $item->get_taxes() ) {
+			return $tax_rates;
+		}
+
+		$tax_rate_ids = array_keys( $item->get_taxes()['total'] );
+
+		if ( ! $tax_rate_ids ) {
+			return $tax_rates;
+		}
+
+		$tax_inclusive = wc_prices_include_tax();
+
+		foreach ( $subscription->get_taxes() as $tax ) {
+			if ( in_array( $tax->get_rate_id(), $tax_rate_ids, true ) ) {
+				$tax_rates[] = [
+					'display_name' => $tax->get_name(),
+					'inclusive'    => $tax_inclusive,
```

Comment:
Maybe we can just use `wc_prices_include_tax()` here instead of creating a variable for this?

Reply:
Ah. I completely missed that. Good call. In that case, nevermind 🙈